File: JFFliter/JFFliter/components/JFFliterSectionHeader.m
//
//  JFFliterSectionHeader.m
//  JFTX_WuHanPJJY
//
//  Created by geeksprite on 2017/7/26.
//  Copyright © 2017年 jf. All rights reserved.
//

#import <Masonry.h>
#import "JFFliterSectionHeader.h"
#import "JFFliterAppearanceManager.h"

@interface JFFliterSectionHeader ()
{
    UIView *_headerView;
    UILabel *_headerLabel;
}

@end

@implementation JFFliterSectionHeader

#pragma mark - Init
- (instancetype)initWithReuseIdentifier:(NSString *)reuseIdentifier {
    if (self = [super initWithReuseIdentifier:reuseIdentifier]) {
        [self didInit];
    }
    return self;
}

#pragma mark - Configure Views
- (void)layoutSubviews {
    [super layoutSubviews];
    if (self.width > 0.00001) {
        [self didLayoutSubviews];
    }
}

- (void)didInit {
    self.contentView.backgroundColor = [UIColor clearColor];
    [self.headerView addSubview:self.headerLabel];
    [self addSubview:self.headerView];
}

- (void)didLayoutSubviews {
    [self.headerLabel sizeToFit];
    [self.headerView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.right.top.equalTo(self);
        make.height.mas_equalTo(self.height);
    }];

    [self.headerLabel mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.equalTo(self.headerView).offset([JFFliterAppearanceManager sharedManager].sectionPaddingSize.width);
        make.top.equalTo(self.headerView).offset([JFFliterAppearanceManager sharedManager].sectionPaddingSize.height);
    }];
}

#pragma mark - Getter && Setter
- (UIView *)headerView {
    if (!_headerView) {
        _headerView= [[UIView alloc] init];
    }
    return _headerView;
}

- (UILabel *)headerLabel {
    if (!_headerLabel) {
        _headerLabel = [[UILabel alloc] init];
        _headerLabel.textAlignment = NSTextAlignmentLeft;
        _headerLabel.font = [UIFont systemFontOfSize:[FontUtil fontSize:11]];
        _headerLabel.textColor = UIColorWithHEX(@"#828282");
    }
    return _headerLabel;
}

@end

File: GGSwift/JFFliter/Components/JFFliterSectionHeader.swift
//
//  JFFliterSectionHeader.swift
//  GGSwift
//
//  Created by geeksprite on 2017/8/1.
//  Copyright © 2017年 聚风天下有限公司. All rights reserved.
//

import UIKit

class JFFliterSectionHeader: UITableViewHeaderFooterView {
    
    override init(reuseIdentifier: String?) {
        super.init(reuseIdentifier: reuseIdentifier)
        self.didInit()
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        guard self.width > 0.00001 else {
            return
        }
        self.didLayoutSubviews()
    }
    
    open func didInit() {
        self.backgroundColor = UIColor.clear
        self.headerView.addSubview(self.headerLabel)
        self.addSubview(self.headerView)

    }
    
    open func didLayoutSubviews() {
        self.headerLabel.sizeToFit()
        self.headerView.frame = self.bounds
        self.headerLabel.frame.origin = CGPoint.init(x: JFFliterAppearceManager.shared.sectionPaddingSize.width, y: JFFliterAppearceManager.shared.sectionPaddingSize.height)
        
    }

    open lazy var headerLabel: UILabel = {
        let label = UILabel()
        label.textAlignment = .left
        label.font = UIFont.systemFont(ofSize: FontUtil.fontSize(size: 11))
        label.textColor = UIColor.colorWithHexString(hex: "#828282")
        return label
    }()
    
    open lazy var headerView: UIView = {
        return UIView()
    }()

}
